.github/workflows/pr.yaml

```
@@ -317,6 +317,39 @@ jobs:
+    name: frontend-api-surface-compliance-check
+    needs: [changed-files, sglang-build]
+    # Narrower than sibling sglang-* jobs: this validates Dynamo's frontend
+    # HTTP surface, not sglang specifically. Fire on any filter that can move
+    # that surface:
+    #   core     -> lib/**, tests/**, container/** (wire-shape fix lands here)
+    #   frontend -> components/src/dynamo/frontend/**
+    #   sglang   -> backend vehicle — backend changes can shift real traffic
+    # sglang-build stays in `needs` because the test spins up sglang as the
+    # traffic source; no sglang runtime image means the test can't run, so
+    # the natural skip cascade via `needs` is the behaviour we want.
+    if: needs.changed-files.outputs.core == 'true' || needs.changed-files.outputs.frontend == 'true' || needs.changed-files.outputs.sglang == 'true'
```

actionlint (1.7.12)

[error] property "frontend" is not defined in object type {builder_name: string; core: string; deploy: string; operator: string; planner: string; sglang: string; trtllm: string; vllm: string}

(expression)